cf-check/backup.c

```diff
@@ -30,28 +32,38 @@ const char *create_backup_dir()
     {
         if (errno != EEXIST)
         {
-            printf("Could not create directory '%s' (%s)\n", backup_root, strerror(errno));
+            Log(LOG_LEVEL_ERR,
+                "Could not create directory '%s' (%s)",
+                backup_root,
+                strerror(errno));
             return NULL;
         }
     }
 
     time_t ts = time(NULL);
-    if (ts == (time_t) -1)
+    if (ts == (time_t)-1)
     {
-        printf("Could not get current time\n");
+        Log(LOG_LEVEL_ERR, "Could not get current time");
         return NULL;
     }
 
-    const int n = snprintf(backup_dir, PATH_MAX, "%s%jd/", backup_root, (intmax_t) ts);
+    const int n =
+        snprintf(backup_dir, PATH_MAX, "%s%jd/", backup_root, (intmax_t)ts);
     if (n >= PATH_MAX)
     {
-        printf("Backup path too long: %jd/%jd\n", (intmax_t) n, (intmax_t) PATH_MAX);
+        Log(LOG_LEVEL_ERR,
+            "Backup path too long: %jd/%jd",
+            (intmax_t)n,
+            (intmax_t)PATH_MAX);
```

Comment:
at first I was confused by the use of intmax_t. That is the largest signed integer type but wasn't sure why you used it here. So please give me a hint and educate me. :)

Reply:
Note that this change is only a formatting change, so I shouldn't change anything here.

When you don't have an appropriate `%` specifier, the best approach is to cast to `uintmax_t` or `intmax_t` and use `%ju` or `%jd`. In this case it was probably because I don't know what type `PATH_MAX` will be on all systems. (`n` is just an `int` and could be printed using `%d`).